marklogic-client-api/src/main/java/com/marklogic/client/DatabaseClientFactory.java

```diff
@@ -973,13 +974,42 @@ public CertificateAuthContext(SSLContext context, SSLHostnameVerifier verifier,
      *           UnrecoverableKeyException.
      */
     public CertificateAuthContext(String certFile, X509TrustManager trustManager)
-        throws CertificateException, IOException,
-        UnrecoverableKeyException, KeyManagementException {
+        	throws CertificateException, IOException, UnrecoverableKeyException, KeyManagementException {
         this.certFile = certFile;
         this.trustManager = trustManager;
         this.certPassword = "";
         this.sslContext = createSSLContext();
-      }
+	}
```

Comment:
Have the above constructor invoke the next one - e.g. `this(certFile, trustManager, certPassword, null)` - and the next one will apply the logic of defaulting a null password to `""` and defaulting the protocol value if null.